sequencer/src/l1_client.rs

```diff
@@ -276,140 +163,70 @@ async fn get_finalized_block<P: JsonRpcClient>(
         setup_logging();
         setup_backtrace();
 
-        let test_duration = Duration::from_secs(30);
-        let test_interval = Duration::from_millis(500);
         let anvil = AnvilOptions::default()
             .block_time(Duration::from_secs(1))
             .spawn()
             .await;
-        let l1_client = L1ClientOptions::with_url(anvil.ws_url())
-            .start()
-            .await
-            .unwrap();
-        let anvil_client = Provider::try_from(anvil.url().to_string()).unwrap();
-
-        let mut prev_head = 0;
-        let mut prev_finalized: Option<L1BlockInfo> = None;
 
-        let start = Instant::now();
-        while start.elapsed() < test_duration {
-            let snapshot = l1_client.snapshot().await;
+        let l1_client = L1Client::new(anvil.url());
+        let provider = l1_client.provider;
 
-            let expected_head = anvil_client.get_block_number().await.unwrap().as_u64();
-            let expected_finalized = anvil_client
-                .get_block(BlockNumber::Finalized)
-                .await
-                .unwrap();
+        let expected_head = provider.get_block_number().await.unwrap().as_u64();
+        let expected_finalized = provider.get_block(BlockNumber::Finalized).await.unwrap();
 
-            // The snapshotted head should be within one of the observed head, it may be one block
-            // behind if the head just changed and the snapshot hasn't updated yet.
-            assert!(
-                expected_head == snapshot.head || expected_head == snapshot.head + 1,
-                "snapshotted head {} does not match expected head {}",
-                snapshot.head,
-                expected_head
-            );
-
-            // And same for the finalized block, unless there is no finalized block.
-            if let Some(expected_finalized) = expected_finalized {
-                let expected_finalized_number = expected_finalized.number.unwrap().as_u64();
-                let snapshot_finalized = snapshot.finalized.unwrap();
-                assert!(
-                    expected_finalized_number == snapshot_finalized.number
-                        || expected_finalized_number == snapshot_finalized.number + 1,
-                    "snapshotted finalized {} does not match expected finalized {}",
-                    snapshot_finalized.number,
-                    expected_finalized_number
-                );
-                if expected_finalized_number == snapshot_finalized.number {
-                    assert_eq!(expected_finalized.hash.unwrap(), snapshot_finalized.hash);
-                    assert_eq!(expected_finalized.timestamp, snapshot_finalized.timestamp);
-                    tracing::info!("L1 finalized block fully synchronized");
-                }
-            } else {
-                tracing::info!("no finalized L1 block yet");
-                assert_eq!(snapshot.finalized, None);
-            }
+        assert_eq!(7, expected_head);
```

Comment:
Where does 7 come from?

Reply:
Not sure. I just observed that it always appeared to be be 7.

Reply:
this was also removed in ef28a828d766970c14c8d2b42b1b2afbadd4347e